core/matrix/ell.cpp

```diff
@@ -101,6 +101,50 @@ size_type calculate_max_nnz_per_row(
 }  // namespace
 
 
+template <typename ValueType, typename IndexType>
+Ell<ValueType, IndexType>& Ell<ValueType, IndexType>::operator=(
+    const Ell& other)
+{
+    if (&other != this) {
+        EnableLinOp<Ell>::operator=(other);
+        values_ = other.values_;
+        col_idxs_ = other.col_idxs_;
+        num_stored_elements_per_row_ = other.num_stored_elements_per_row_;
+        stride_ = other.stride_;
+    }
+    return *this;
+}
+
+
+template <typename ValueType, typename IndexType>
+Ell<ValueType, IndexType>& Ell<ValueType, IndexType>::operator=(Ell&& other)
+{
+    if (&other != this) {
+        EnableLinOp<Ell>::operator=(std::move(other));
+        values_ = std::move(other.values_);
+        col_idxs_ = std::move(other.col_idxs_);
+        num_stored_elements_per_row_ =
+            std::exchange(other.num_stored_elements_per_row_, 0);
+        stride_ = std::exchange(other.stride_, 0);
```

Comment:
Aren't num_stored_elements_per_row_ and stride_ zero from original move?

Reply:
They are members directly inside the Ell type, thus they need to be handled inside its assignment operator. Primitive types don't reset on move assignment afaik.